app/views/contexts/summary_context.py

```diff
@@ -96,25 +92,21 @@ def build_summary_item(
         summary = section_summary_context(
             view_submitted_response=self.view_submitted_response, return_to=return_to
         )["summary"]
-        groups: list = summary.get("groups", [])
-
-        for group in groups:
-            group["section_title"] = summary["title"]
-            break
-
-        return groups
-
+        title: str = summary["title"]
 
-def set_unique_group_ids_and_titles(groups: list[dict]) -> list[dict]:
-    checked_ids = set()
-    id_value = 0
+        if groups := summary.get("sections", {}).get("groups", []):
+            if final_groups := [group for group in groups if len(group["blocks"]) > 0]:
+                self.summaries[title] = final_groups
```

Comment:
in `section_summary_context.py` the `title_for_location` might be null (for instance in `test_calculated_summary.json`) which passed through to this and we get None as the key for the dictionary entry - this seems to be as it doesn't have a section title/repeating section title etc, so in addition to above comment I think theres another issue with which title is being picked up

Reply:
Changed the logic here as per Mebin's comment above 👍